composer/loggers/tensorboard_logger.py

```diff
@@ -115,11 +131,32 @@ def _flush(self, logger: Logger):
         assert self.writer.file_writer is not None
         file_path = self.writer.file_writer.event_writer._file_name
 
+        # If no writes have happened since the last flush, then file_path won't exist, so
+        # we should skip doing flushing and skip filing the artifact since it will error
+        # out anyway (given that the file_path doesn't exist).
+        if not Path(file_path).exists():
+            return
+
+        assert self.writer is not None
```

Comment:
given that we do `        assert self.writer.file_writer is not None` above, is this still neccessary?

Also, I see that the writer has already been flushed in lines 128-129, so is this second flush necessary given that nothing new had changed about the writer?

Reply:
sorry this was just a mistaken duplicate line when transposing the old PR to remove flush_secs. fixed. good catch